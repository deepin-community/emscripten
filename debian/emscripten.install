Xtypes/package.json Xtypes/*.d.ts /usr/share/nodejs/@types/emscripten
debian/bin/wrapper debian/bin/wrapper-tools debian/tmp/* debian/.emscripten /usr/share/emscripten
