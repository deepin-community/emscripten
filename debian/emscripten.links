/usr/share/emscripten/wrapper /usr/bin/em++
/usr/share/emscripten/wrapper /usr/bin/em-config
/usr/share/emscripten/wrapper /usr/bin/emar
/usr/share/emscripten/wrapper /usr/bin/embuilder
/usr/share/emscripten/wrapper /usr/bin/emcc
/usr/share/emscripten/wrapper /usr/bin/emcmake
/usr/share/emscripten/wrapper /usr/bin/emconfigure
/usr/share/emscripten/wrapper /usr/bin/emmake
/usr/share/emscripten/wrapper /usr/bin/emprofile
/usr/share/emscripten/wrapper /usr/bin/emranlib
/usr/share/emscripten/wrapper /usr/bin/emrun
/usr/share/emscripten/wrapper /usr/bin/emscons
/usr/share/emscripten/wrapper /usr/bin/emsize
/usr/share/emscripten/wrapper-tools /usr/bin/emdump
/usr/share/emscripten/wrapper-tools /usr/bin/emdwp
/usr/share/emscripten/wrapper-tools /usr/bin/emnm
/usr/share/emscripten/wrapper-tools /usr/bin/emprofile
