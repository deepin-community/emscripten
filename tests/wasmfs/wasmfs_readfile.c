/*
 * Copyright 2021 The Emscripten Authors.  All rights reserved.
 * Emscripten is available under two separate licenses, the MIT license and the
 * University of Illinois/NCSA Open Source License.  Both these licenses can be
 * found in the LICENSE file.
 */

#include <assert.h>
#include <emscripten.h>
#include <errno.h>
#include <fcntl.h>
#include <stdio.h>
#include <string.h>
#include <unistd.h>

int main() {
  int fd = open("/test", O_CREAT, S_IRGRP);
  const char* msg = "Success\n";

  errno = 0;
  write(fd, msg, strlen(msg));
  EM_ASM({
    var output = FS.readFile("/test", {encoding : 'utf8'});
    out(output);
  });

  EM_ASM({
    try {
      var output = FS.readFile("/no-exist", {encoding : 'utf8'});
    } catch (err) {
    }
  });

  EM_ASM({
    try {
      var output = FS.readFile("", {encoding : 'utf8'});
    } catch (err) {
    }
  });
}
