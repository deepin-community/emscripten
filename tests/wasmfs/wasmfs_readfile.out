Success

Fatal error in FS.readFile
Aborted(native code called abort())
Fatal error in FS.readFile
Aborted(native code called abort())
